src/lib/components/TextInput/TextInput.tsx

```diff
@@ -74,16 +78,25 @@ export const TextInput = forwardRef<HTMLInputElement, TextInputProps>(
         <div className={classNames(theme.base, className)}>
           {addon && <span className={theme.addon}>{addon}</span>}
           <div className={theme.field.base}>
-            {Icon && (
+            {Icon ? (
               <div className={theme.field.icon.base}>
                 <Icon className={theme.field.icon.svg} />
               </div>
-            )}
-            {RightIcon && (
+            ) : renderIcon ? (
+              <div data-testid="rendered-icon" className={theme.field.icon.base}>
+                {renderIcon(theme.field.icon.svg)}
+              </div>
+            ) : null}
```

Comment:
I would not use `null` here, instead, I would replace the `renderIcon` with `renderIcon && (...)` - same for `renderRightIcon`

Except by that, it lgtm.

Reply:
I tried something like that and was getting some weird behavior so I just went with the way I knew would work, but I will definitely try and figure it out.

Reply:
Thanks. @rtraficante.

Reply:
No problem! @rluders